logging/logging.go

```diff
@@ -57,11 +58,37 @@ func DisableLogging() {
 
 // EnableStructuredLogs replaces all logging functions with structured logging
 // variants.
-func EnableStructuredLogs() (func(), error) {
-	logger, err := zap.NewProduction()
-	if err != nil {
-		return func() {}, err
+func EnableStructuredLogs(logDebugStdout bool) (func(), error) {
+	// Define level-handling logic.
+	highPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl >= zapcore.ErrorLevel
+	})
+	lowPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl < zapcore.ErrorLevel
+	})
+
+	// Lock consoles
+	consoleErrors := zapcore.Lock(os.Stderr)
+
+	var consoleDebugging zapcore.WriteSyncer
+	if logDebugStdout {
+		consoleDebugging = zapcore.Lock(os.Stdout)
+	} else {
+		consoleDebugging = consoleErrors
 	}
+
+	// Create JSON encoder using opinionated production preset
+	consoleEncoder := zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig())
+
+	// Join the outputs, encoders, and level-handling functions into
+	// zapcore.Cores, then tee them together.
+	core := zapcore.NewTee(
+		zapcore.NewCore(consoleEncoder, consoleErrors, highPriority),
+		zapcore.NewCore(consoleEncoder, consoleDebugging, lowPriority),
+	)
+
+	// Create logger, also add caller and stacktrace
```

Comment:
nit: Same for these comments. They can be either removed or updated to add context on why we are doing the specific line.

Reply:
Added a link in the comments to the official example on which this is based.